www/officers/coi.cgi

```diff
@@ -0,0 +1,201 @@
+  centered_name = "#{name}".center(50, '_')
+  centered_date ="#{timestamp}".center(51, '_')
+  filled_signature_block = signature_block
+    .gsub('Signed: __', ('Signed: ' + centered_name))
+    .gsub('Date: __'  , (  'Date: ' + centered_date))
+  template + filled_signature_block
+end
+
+_html do
+  _body? do
+    _whimsy_body(
+    title: PAGETITLE,
+    related: {
+      'http://www.apache.org/foundation/records/minutes/2020/board_minutes_2020_03_18.txt'  =>
+        'Conflict of Interest Resolution Board minutes',
+      COI_CURRENT_TEMPLATE_URL => 'Conflict of Interest Resolution',
+    },
+    helpblock: -> {
+      _p do
+        _ 'This page allows officers to sign their Conflict of Interest annual affirmation.'
+      end
+      if _.get?
+      _p 'The following are currently required to affirm the Conflict of Interest:'
+      IDS.each do |id|
+       affirmer = ASF::Person.find(id)
+       _ "#{affirmer.cn} (#{affirmer.id}) "
+      end
+      _p
+      _b 'Signers for this year:'
+      SIGNERS.each do |signer, signerfile|
+        _a signer, href: "#{COI_CURRENT_URL}/#{signerfile}"
+      end
+      _p
+      _b 'Nonsigners for this year:'
+      NONSIGNERS.each do |nonsigner|
+        _ nonsigner
+      end
+      _p
+      _p "You are signed in as #{USERNAME} (#{USERID}) at #{current_timestamp}."
+      _p {_ "You are ";_b "#{not_required_message}";_ " to affirm the Conflict of Interest policy for this year."}
+      _p {_ "You ";_b "#{have_affirmed_message}";_  "the Conflict of Interest policy for this year."}
+      if  USER_AFFIRMATION_FILE
+        _a "Your Conflict of Interest affirmation",
+          href: "#{COI_CURRENT_URL}/#{USER_AFFIRMATION_FILE}"
+      end
+      if USER_IS_REQUIRED_BUT_NOT_AFFIRMED
+        _p {_b "You are invited to sign the affirmation below"}
+      end
+    end
+    }
+    ) do
+      if _.get?
+        if USER_IS_REQUIRED_BUT_NOT_AFFIRMED
+          _whimsy_panel(PANEL_MESSAGE, style: 'panel-success') do
+            affirmed = get_affirmed_template($USER, $PASSWORD, USERNAME,  current_timestamp)
+            affirmed.each_line do |line|
+              _p line
+            end
+            _form.form_horizontal method: 'post' do
+              _div.form_group do
+                _div.col_sm_offset_1.col_sm_10 do
+                  _input.btn.btn_default type: 'submit',
+                    value: 'Sign your Conflict of Interest Affirmation'
+                end
+              end
+            end
+          end
+        end
+      else # POST
+        _whimsy_panel('Sign Conflict of Interest Affirmation - Session Transcript',
+            style: 'panel-success') do
+          _div.transcript do
+            emit_post(_)
+          end
+        end
+      end
+    end
+  end
+end
+
+# Emit a record of a user's submission - POST
+def emit_post(_)
+  # The only information in the POST is $USER and $PASSWORD
+  user = ASF::Person.find($USER)
```

Comment:
This is a duplicate of line 16

Reply:
I've removed this line and added new constant: USERMAIL which was the reason this function needed user.